### DISCUSSION TITLE
Is it possible to change the column order?

### DISCUSSION THREAD

**Discussion by Vadorequest** [supabase/supabase#622] [Questions] [answered]
Column are ordered in the order they are created.

Is there a way to override this? The most obvious way using drag & drop doesn't work.

Looking around, it doesn't seem to be something easy/possible with native SQL, so I guess that could be done in the UI but might not be implemented yet?

https://stackoverflow.com/questions/285733/how-do-i-alter-the-position-of-a-column-in-a-postgresql-database-table

**Comment by itechnica** [comment]
Postgres doesn't support column reordering natively; the only way to do a 'real' reorder is to dump the table, alter the dump, and recreate the table (or some variation of that process).

Depending on your reason for wanting to reorder the table, you could simply create a VIEW with the columns in the required order.

If it was to be done in the UI, it could only be a 'fake' reordering, and it would be painful to implement; probably something along the lines of storing the desired UI display order in a table, which would result in slowing down the rendering of the table.

I suppose I should have started by asking, "What is the reason for wanting to reorder the columns?", as that would have helped to narrow down the answer!

**Comment by cdedreuille** [comment]
I would love to have this feature as well. Many times I'm structuring my data in one way and end up by having the most important pieces of information at the end of the table. With column ordering, you don't have to think about it straight away.

**Comment by silentworks** [comment]
I use TablePlus as my SQL editor and it has a data table view where you can reorder the columns, this might be helpful for you since it doesn't exist in the Supabase dashboard.

**Comment by inian** [comment, marked as answer]
This is supported now. The column order is stored in your localStorage though and doesn't persist across different browser sessions.

**Reply by paksa92** [reply]
Can we have this in all views with database columns? For example, at `/dashboard/project/[project_id]/database/tables/[table_id]` or in the sidesheet when editing a table?
